oap-server/server-cluster-plugin/cluster-consul-plugin/src/main/java/org/apache/skywalking/oap/server/cluster/plugin/consul/ConsulCoordinator.java

```diff
@@ -49,22 +56,32 @@ public ConsulCoordinator(ClusterModuleConsulConfig config, Consul client) {
 
     @Override
     public List<RemoteInstance> queryRemoteNodes() {
-        HealthClient healthClient = client.healthClient();
-
-        // Discover only "passing" nodes
-        List<ServiceHealth> nodes = healthClient.getHealthyServiceInstances(serviceName).getResponse();
-
         List<RemoteInstance> remoteInstances = new ArrayList<>();
-        if (CollectionUtils.isNotEmpty(nodes)) {
-            nodes.forEach(node -> {
-                if (!Strings.isNullOrEmpty(node.getService().getAddress())) {
-                    Address address = new Address(node.getService().getAddress(), node.getService().getPort(), false);
-                    if (address.equals(selfAddress)) {
-                        address.setSelf(true);
+        try {
+            HealthClient healthClient = client.healthClient();
+            // Discover only "passing" nodes
+            List<ServiceHealth> nodes = healthClient.getHealthyServiceInstances(serviceName).getResponse();
+            if (CollectionUtils.isNotEmpty(nodes)) {
+                nodes.forEach(node -> {
+                    if (!Strings.isNullOrEmpty(node.getService().getAddress())) {
+                        Address address = new Address(node.getService().getAddress(), node.getService().getPort(), false);
+                        if (address.equals(selfAddress)) {
+                            address.setSelf(true);
+                        }
+                        remoteInstances.add(new RemoteInstance(address));
```

Comment:
If you found `127.0.0.1` or `localhost` exists in the node list(list#size > 1), then the node should be labeled as unhealthy.
FYI @hanahmily @kezhenxu94 Make sense?

Reply:
Same thing should be checked in every coordinator implementation. I noted that the users usually set up the cluster mode incorrectly, then, blame SkyWalking insert duplicated data or statistic not accurate.

Reply:
got it

Reply:
Done, current implementation will ignore coordinator  standalone mode